src/components/views/rooms/RoomHeader.tsx

```diff
@@ -48,22 +51,287 @@ import { BetaPill } from "../beta/BetaCard";
+    room: Room;
+    busy: boolean;
+    setBusy: (value: boolean) => void;
+    behavior: DisabledWithReason | "legacy" | "element" | "legacy_or_element";
+}
+
+/**
+ * Button for starting video calls, supporting both legacy 1:1 calls, Jitsi
+ * widgets, and native group calls. If multiple calling options are available,
+ * this shows a menu to pick between them.
+ */
+const VideoCallButton: FC<VideoCallButtonProps> = ({ room, busy, setBusy, behavior }) => {
+    const [menuOpen, buttonRef, openMenu, closeMenu] = useContextMenu();
+
+    const startLegacyCall = useCallback(async () => {
+        setBusy(true);
+        await LegacyCallHandler.instance.placeCall(room.roomId, CallType.Video);
+        setBusy(false);
+    }, [setBusy, room]);
+
+    const startElementCall = useCallback(() => {
+        setBusy(true);
+        defaultDispatcher.dispatch<ViewRoomPayload>({
+            action: Action.ViewRoom,
+            room_id: room.roomId,
+            view_call: true,
+            metricsTrigger: undefined,
+        });
+        setBusy(false);
+    }, [setBusy, room]);
+
+    const { onClick, tooltip, disabled } = useMemo(() => {
+        if (behavior instanceof DisabledWithReason) {
+            return {
+                onClick: () => {},
+                tooltip: behavior.reason,
+                disabled: true,
+            };
+        } else if (behavior === "legacy") {
+            return {
+                onClick: async (ev: ButtonEvent) => {
+                    ev.preventDefault();
+                    await startLegacyCall();
+                },
+                disabled: false,
+            };
+        } else if (behavior === "element") {
+            return {
+                onClick: async (ev: ButtonEvent) => {
+                    ev.preventDefault();
+                    startElementCall();
+                },
+                disabled: false,
+            };
+        } else {
+            return {
+                onClick: async (ev: ButtonEvent) => {
+                    ev.preventDefault();
+                    openMenu();
+                },
+                disabled: false,
+            };
+        }
+    }, [behavior, startLegacyCall, startElementCall, openMenu]);
+
+    const onJitsiClick = useCallback(async (ev: ButtonEvent) => {
+        ev.preventDefault();
+        closeMenu();
+        await startLegacyCall();
+    }, [closeMenu, startLegacyCall]);
+
+    const onElementClick = useCallback((ev: ButtonEvent) => {
+        ev.preventDefault();
+        closeMenu();
+        startElementCall();
+    }, [closeMenu, startElementCall]);
+
+    let menu: JSX.Element | null = null;
+    if (menuOpen) {
+        const buttonRect = buttonRef.current!.getBoundingClientRect();
+        menu = <IconizedContextMenu {...aboveLeftOf(buttonRect)} onFinished={closeMenu}>
+            <IconizedContextMenuOptionList>
+                <IconizedContextMenuOption label={_t("Video call (Jitsi)")} onClick={onJitsiClick} />
```

Comment:
```suggestion
                <IconizedContextMenuOption label={_t("Video call")} onClick={onJitsiClick} />
```

This also does 1:1 calls over Matrix, right?

Reply:
Oh, I guess it technically could but the `CallButtons` component only ever passes `"legacy_or_element"` when there are 3+ members. I'll rename the option to `"jitsi_or_element"` to make that more explicit.